File: Python/Python/largest_smallest_array.py
#!/usr/bin/env python3

# How do you find the largest and smallest number in an unsorted integer array?

import random

somearray = [None]* 100

for i in range(100):
    somearray[i] = random.randint(1001)

for i in range(100):
    if i not in somearray:
        print("The missing number is: {}".format(i))

File: C/largest_smallest_array.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define ARRAYSIZE 100

// How do you find the largest and smallest number in an unsorted integer array?


int main(int argc, char* argv[]){
    int i;
    int largest = 0;
    int smallest = 0;

    int somearray[ARRAYSIZE];

    srand(time(0));

    // populate array with random numbers
    for (i = 0; i < ARRAYSIZE; i++){
        somearray[i] = rand() % 1000 + 1;
    }

    for( i = 0; i < ARRAYSIZE; i++){
        if(smallest > somearray[i]){
            smallest = somearray[i];
        }
        if(largest < somearray[i]){
            largest = somearray[i];
        }
    }

    printf("The largest value in the array is: %d\n", largest);
    printf("The smallest value in the array is: %d\n", smallest);
    return 0;
}